community/planning/batch_processor.md

```diff
@@ -0,0 +1,84 @@
+<!--
+  Copyright 2021 Cargill Incorporated
+  Licensed under Creative Commons Attribution 4.0 International License
+  https://creativecommons.org/licenses/by/4.0/
+-->
+
+# Batch Processor
+
+The batch processor is a proposed component meant to monitor a database that
+holds batched transactions and send them to either a Hyperledger Sawtooth or
+Splinter network. Batch processors are meant to be run as part of a cluster with
+other batch processors in order to provide redundancy or reduce the time it
+takes for batches to be processed.
+
+## Batch Submission Algorithm
+
+A batch processor sleeps for a configurable number of seconds before waking
+up and performing the following actions.
+
+1. A processor will query the database for a configurable number of unclaimed
+batches (the `claim_limit`). An unclaimed batch is a batch whose submitted
+status is `false`, and its `claim_expires` time stamp is `NULL` or less than the
+`CURRENT_TIMESTAMP` of the database.
+```
+SELECT * FROM batches
+  WHERE submitted = false
+    AND (claim_expirers is NULL
+      OR claim_expires < CURRENT_TIMESTAMP) LIMIT {claim_limit};
+```
+
+2. The processor will then update the `claim_expires` column to lay claim to
```

Comment:
What are reasonable values for claim_expires and the sleep interval?

Reply:
The current default is 10 seconds.